src/api/api.service.ts

```diff
@@ -1,22 +1,36 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotAcceptableException } from '@nestjs/common';
 import { BullService } from 'src/bull/bull.service';
 import { PrismaService } from 'src/prisma/prisma.service';
 import { SearchResult } from './types';
 import { Job } from 'bullmq';
 
 @Injectable()
 export class ApiService {
+  private keyword_limit = 100;
   constructor(
     private primsaService: PrismaService,
     private bullservice: BullService,
-  ) { }
+  ) {}
 
   findById(id: number) {
     return this.primsaService.search_results.findFirst({
       where: { id },
     });
   }
 
+  async uploadCSV(entities, user_id) {
+    if (entities.list.length > this.keyword_limit) {
+      throw new NotAcceptableException('exceed the limit of keyword per csv');
```

Comment:
How about using the built-in [NotAcceptableException](https://docs.nestjs.com/exception-filters#built-in-http-exceptions) as 422 is more common in such cases than 406 (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406)

Reply:
I'm not entirely clear on this one. I believe I might already be utilizing the built-in `NotAcceptableException` from NestJS.